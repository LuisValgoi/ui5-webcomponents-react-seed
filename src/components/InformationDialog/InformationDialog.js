import React, { useEffect } from 'react';
import { useTranslation } from 'react-i18next';
import { spacing } from '@ui5/webcomponents-react-base';
import { Icon } from '@ui5/webcomponents-react/lib/Icon';
import { Button } from '@ui5/webcomponents-react/lib/Button';
import { ButtonDesign } from '@ui5/webcomponents-react/lib/ButtonDesign';
import { Dialog } from '@ui5/webcomponents-react/lib/Dialog';
import { FlexBox } from '@ui5/webcomponents-react/lib/FlexBox';
import { FlexBoxDirection } from '@ui5/webcomponents-react/lib/FlexBoxDirection';
import { FlexBoxAlignItems } from '@ui5/webcomponents-react/lib/FlexBoxAlignItems';
import { FlexBoxJustifyContent } from '@ui5/webcomponents-react/lib/FlexBoxJustifyContent';
import { Text } from '@ui5/webcomponents-react/lib/Text';

const KEYBOARD_KEYS = {
  ESCAPE: 27,
};

const style = {
  warning: {
    width: '1.5rem',
    height: '1.5rem',
    color: '#feb60a',
  },
  error: {
    width: '1.5rem',
    height: '1.5rem',
    color: '#ff5254',
  },
  information: {
    width: '1.5rem',
    height: '1.5rem',
    color: 'black',
  },
  text: {
    lineHeight: '20px',
  },
};

const _getHeaderIcon = (type) => {
  switch (type) {
    case Type.Warning:
      return _getHeaderWarningIcon();
    case Type.Error:
      return _getHeaderErrorIcon();
    default:
      return _getHeaderInfoIcon();
  }
};

const _getHeaderWarningIcon = () => {
  return <Icon name="message-warning" style={style.warning} />;
};

const _getHeaderErrorIcon = () => {
  return <Icon name="message-error" style={style.error} />;
};

const _getHeaderInfoIcon = () => {
  return <Icon name="message-information" style={style.information} />;
};

const _handleAvoidEscapeClosing = (avoidEscapeClose) => {
  document.addEventListener(
    'keydown',
    (e) => {
      if (e.keyCode === KEYBOARD_KEYS.ESCAPE && avoidEscapeClose) {
        e.stopPropagation();
      }
    },
    true,
  );
};

const InformationDialog = ({ dialogRef, avoidEscapeClose, headerText, innerText, closeButtonText, children, onClose, type }) => {
  const { t } = useTranslation();

  useEffect(() => {
    _handleAvoidEscapeClosing(avoidEscapeClose);
  });

  const _onClose = () => {
    onClose && onClose();
    if (dialogRef.current) {
      dialogRef.current.close();
    }
  };

  const _getFooter = () => {
    return (
      <FlexBox alignItems={FlexBoxAlignItems.Center} direction={FlexBoxDirection.RowReverse} style={spacing.sapUiTinyMargin}>
        <Button design={ButtonDesign.Transparent} onClick={_onClose}>
          {closeButtonText ? closeButtonText : t('app.generics.close')}
        </Button>
      </FlexBox>
    );
  };

  const _getHeader = () => {
    return (
      <FlexBox alignItems={FlexBoxAlignItems.Center} justifyContent={FlexBoxJustifyContent.Center} style={spacing.sapUiContentPadding}>
        {_getHeaderIcon(type)}
        <Text tooltip={headerText} wrapping style={{ ...spacing.sapUiTinyMarginBegin, ...style.text }}>
          {headerText}
        </Text>
      </FlexBox>
    );
  };

  return (
    <Dialog ref={dialogRef} slot="header" header={_getHeader()} footer={_getFooter()} onAfterClose={_onClose} data-testid="information-dialog">
      <div style={{ ...spacing.sapUiContentPadding }}>
        <FlexBox direction={FlexBoxDirection.Column}>
          {innerText ? (
            <Text tooltip={innerText} wrapping style={{ ...spacing.sapUiTinyMarginBegin, ...style.text }}>
              {innerText}
            </Text>
          ) : (
            children
          )}
        </FlexBox>
      </div>
    </Dialog>
  );
};

export default InformationDialog;

export const Type = {
  Warning: 'WARNING',
  Error: 'ERROR',
  Info: 'INFO',
};
