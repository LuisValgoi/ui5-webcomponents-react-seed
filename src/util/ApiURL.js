export default {
  GET_USER_LOGGED: '/v1/user/logged',
  GET_TODO_LIST: '/v1/todo/all',
  GET_TODO_BY_ID: '/v1/todo/:id'
};
