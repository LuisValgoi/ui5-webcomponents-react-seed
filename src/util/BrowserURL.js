export default {
  HOME: '/',
  TODO_LIST: '/todo/list',
  TODO_DETAIL: '/todo/:id/detail',
  NOT_FOUND: '/notFound',
  BUGGY: '/buggy',
  ANY: '/*'
};
