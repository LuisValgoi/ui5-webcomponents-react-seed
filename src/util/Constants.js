export default {
  REACT_QUERY: {
    CODES: {
      SUCCESS: 'success',
      LOADING: 'loading',
      ERROR: 'error',
    },
    KEYS: {
      RQ_GET_USER_LOGGED: 'RQ_GET_USER_LOGGED',
      RQ_GET_TODO_LIST: 'RQ_GET_TODO_LIST',
      GET_TODO_BY_ID: 'GET_TODO_BY_ID',
    },
  },
  SEED: {
    SELECTED_THEME: 'SELECTED_THEME',
  },
};
