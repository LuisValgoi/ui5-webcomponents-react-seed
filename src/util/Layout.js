import React from 'react';
import { Text } from '@ui5/webcomponents-react/lib/Text';

export const TextWrapper = ({ children, isOnHeader, style }) => {
  return (
    <Text
      tooltip={children}
      wrapping
      className={isOnHeader && 'text-header-color'}
      style={{ ...style, lineHeight: '20px' }}
    >
      {children}
    </Text>
  );
};
