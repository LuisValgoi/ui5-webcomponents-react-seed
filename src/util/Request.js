import axios from 'axios';

const Request = axios.create({
  baseURL: process.env.REACT_APP_API_URL
});

export default Request;
