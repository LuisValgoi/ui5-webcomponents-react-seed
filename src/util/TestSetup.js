import React from 'react';

import { render as rtlRender } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { rest } from 'msw';
import { setupServer } from 'msw/node';

import BrowserProvider from './browser/BrowserProvider';

const render = (ui, { route = BrowserProvider.getUrl('HOME'), ...renderOptions } = {}) => {
  const WrapperProvider = ({ ...props }) => {
    const history = createMemoryHistory({ initialEntries: [route] });
    return (
      <BrowserRouter>
        <Router history={history}>{props.children}</Router>
      </BrowserRouter>
    );
  };

  return rtlRender(ui, { wrapper: WrapperProvider, ...renderOptions });
};

const serverCustom = (apiUrl, data) => {
  return setupServer(
    rest.get(`*${apiUrl}`, (req, res, ctx) => {
      return res(ctx.json(data));
    }),
  );
};

export * from '@testing-library/react';

export { render, serverCustom };
