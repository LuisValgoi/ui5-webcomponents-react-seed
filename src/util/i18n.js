import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

import LanguageDetector from 'i18next-browser-languagedetector';

import translationEN from '../locales/en/translation.json';
import translationPT from '../locales/pt/translation.json';

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    debug: true,
    keySeparator: false,
    interpolation: {
      escapeValue: false,
    },
    resources: {
      en: {
        translation: translationEN,
      },
      pt: {
        translation: translationPT,
      },
    },
    react: {
      useSuspense: false,
    },
  });

export default i18n;
