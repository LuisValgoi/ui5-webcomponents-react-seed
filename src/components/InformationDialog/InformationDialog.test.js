import React from 'react';

import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '../../util/TestSetup';
import InformationDialog, { Type } from './InformationDialog';

describe('InformationDialog.js Test Suite', () => {
  test('should render', () => {
    const dialog = <InformationDialog avoidEscapeClose type={Type.Warning} headerText={'Header text'} closeButtonText={'Close'} innerText={'Inner text'} />;
    render(dialog);
    const infoDialog = screen.getByTestId('information-dialog');
    expect(infoDialog).toBeInTheDocument();
  });

  test('should render child when not inner text is passed', () => {
    const dialog = (
      <InformationDialog avoidEscapeClose type={Type.Warning} headerText={'Header text'} closeButtonText={'Close'}>
        <div data-testid="information-dialog-child"></div>
      </InformationDialog>
    );
    render(dialog);
    const child = screen.getByTestId('information-dialog-child');
    expect(child).toBeInTheDocument();
  });
});
