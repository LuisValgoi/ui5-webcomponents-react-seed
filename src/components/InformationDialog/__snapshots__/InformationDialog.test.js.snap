// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InformationDialog.js Test Suite should match snapshot 1`] = `
<DocumentFragment>
  Dialog {
    "_domRefReadyPromise": Promise {
      "_deferredResolve": [Function],
    },
    "_firePropertyChange": false,
    "_fullyConnected": false,
    "_inDOM": false,
    "_monitoredChildProps": Map {},
    "_shouldInvalidateParent": false,
    "_state": Object {
      "_blockLayerHidden": false,
      "_disableInitialFocus": false,
      "ariaLabel": undefined,
      "default": Array [],
      "footer": Array [],
      "header": Array [],
      "headerText": "",
      "initialFocus": "",
      "onPhone": false,
      "opened": false,
      "preventFocusRestore": false,
      "stretch": false,
    },
    "_upToDate": false,
    "staticAreaItem": StaticAreaItem {
      "_rendered": false,
      "ui5ElementContext": [Circular],
    },
  }
</DocumentFragment>
`;
